backend/app/sql/get_block_zone_assignments_bboxes.sql

```diff
@@ -0,0 +1,33 @@
+CREATE OR REPLACE FUNCTION get_block_assignments_bboxes(document_id UUID, zones INTEGER[])
+RETURNS TABLE (geo_id TEXT, zone INTEGER, bbox BOX2D) AS $$
+DECLARE
+    doc_districtrmap RECORD;
+    sql_query TEXT;
+BEGIN
+    SELECT districtrmap.* INTO doc_districtrmap
+    FROM document.document
+    LEFT JOIN districtrmap
+    ON document.districtr_map_slug = districtrmap.districtr_map_slug
+    WHERE document.document_id = $1;
+
+    IF doc_districtrmap.districtr_map_slug IS NULL THEN
+        RAISE EXCEPTION 'Table name not found for document_id: %', $1;
+    END IF;
+
+    IF doc_districtrmap.child_layer IS NULL THEN
+        RAISE EXCEPTION 'Child layer is NULL for document_id: %. Block-level exports are not supported', $1;
+    ELSE
+        sql_query := format('
```

Comment:
H: This query doesn't filter zones current, which could improve performance. Locally, I'm seeing ~2x gains by adding the `WHERE` clauses here.

```suggestion
 sql_query := format($f$
            SELECT
                assignments.geo_id,
                assignments.zone,
                Box2D(blocks.geometry) AS bbox
            FROM get_block_assignments($1::UUID, $2) assignments
            JOIN gerrydb.%I blocks
            ON blocks.path = assignments.geo_id
            WHERE assignments.zone IS NOT NULL
            AND assignments.zone = ANY($2)
        $f$, doc_districtrmap.child_layer);
```

Reply:
nice